dbt/adapters/athena/impl.py

```diff
@@ -658,6 +665,28 @@ def list_relations_without_caching(self, schema_relation: AthenaRelation) -> Lis
 
         return relations
 
+    def _get_one_catalog_by_relations(
+        self,
+        information_schema: InformationSchema,
+        relations: List[BaseRelation],
+        manifest: Manifest,
+    ) -> agate.Table:
+        """
+        Overwrite of _get_one_catalog_by_relations for Athena, in order to use glue apis.
+        This function is invoked by Adapter.get_catalog_by_relations.
+        """
+        _table_definitions = []
+        for _rel in relations:
+            glue_table_definition = self.get_glue_table(_rel)
+            if glue_table_definition:
+                _table_definition = self._get_one_table_for_catalog(glue_table_definition["Table"], _rel.database)
+                print(_table_definition)
+                _table_definitions.append(_table_definition)
+        table = agate.Table.from_object(_table_definitions)
+        # TODO this doesn't work, must be fixed
+        # return self._join_catalog_table_owners(table, manifest)
```

Comment:
@Jrmyy table contains the output of _get_one_table_for_catalog - actually one entry per column.

I want to use _join_catalog_table_owners to enrich the table with the owner, but somehow I get: 
`E       KeyError: 'table_database'`

_table_definitions is:
```
[{'table_database': 'awsdatacatalog', 'table_schema': 'foo', 'table_name': 'bar', 'table_type': 'table', 'table_comment': '', 'column_name': 'id', 'column_index': 0, 'column_type': 'string', 'column_comment': ''}, {'table_database': 'awsdatacatalog', 'table_schema': 'foo', 'table_name': 'bar', 'table_type': 'table', 'table_comment': '', 'column_name': 'country', 'column_index': 1, 'column_type': 'string', 'column_comment': ''}, {'table_database': 'awsdatacatalog', 'table_schema': 'foo', 'table_name': 'bar', 'table_type': 'table', 'table_comment': '', 'column_name': 'dt', 'column_index': 2, 'column_type': 'date', 'column_comment': ''}]
```